packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -314,14 +314,14 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
       // If the write block failed because the index doesn't exist, it means
       // another instance already completed the legacy pre-migration. Proceed
       // to the next step.
-      if (isTypeof(res.left, 'index_not_found_exception')) {
+      const left = res.left;
+      if (isTypeof(left, 'index_not_found_exception')) {
         return { ...stateP, controlState: 'LEGACY_CREATE_REINDEX_TARGET' };
       } else {
-        // @ts-expect-error TS doesn't correctly narrow this type to never
-        return throwBadResponse(stateP, res);
+        throwBadResponse(stateP, left);
```

Comment:
I assume it was just a mistake that we were previously throwing `res` here instead of `res.left`, and that we aren't losing anything important in our error?

Reply:
In this branch of the conditional, `res` should not contain anything apart from `left`.

I looked at the code that can throw this exception:
`packages/core/saved-objects/core-saved-objects-migration-server-internal/src/actions/set_write_block.ts`

And I can confirm, the only possible _Left_ responses are `IndexNotFound | RetryableEsClientError`, none of them have anything else.
Finally, the `throwBadResponse` only **JSON.stringifies** the received object, so I believe it's completely safe to delete.